### DISCUSSION TITLE
Project Future Direction

### DISCUSSION THREAD

**Discussion by Songtech-0912** [tenacityteam/tenacity-legacy#159] [General]
Hi all! 

I understand that Tenacity has recently faced some serious setbacks. The recent attacks on @cookiengineer are a catastrophe that I think is better left undiscussed - we all know about the horrible events. I believe that for Tenacity to move forward, we need to have a **comprehensive discussion** on the project. With my perspective as a designer and developer, my opinion is to do these things:

* The first thing we have to emphasize is **maximum user privacy**. This is pretty much clear from the onset, we are making Tenacity to be privacy-centric. With that in mind, we need to move forward with "user-respecting" as a major theme - we should rewrite the website to remove all forms of analytics and work with LibreJS, among other things. However, we must be careful in doing this, as we don't want to perpetuate the common myth that "privacy == inconvenience". We must begin
* However, our major obstacle is that we are seen as **amateurish**. To resolve this issue, we need to establish ourselves as a team dedicated to making a professional _and_ user-centric audio editor. We need to bring back some of the polish of Audacity but with the special touch of Tenacity - restyling the website for a different look and feel might be a start, as well as continuing the ongoing discussion on designing a new logo (which I can do). Additionally, we should immediately begin work on the docs - let's keep those in Markdown so that we don't need to run JavaScript for them, and using CSS to handle all the styling.
* The second thing we need to do is to **make our community as welcoming as possible**. Tenacity is in unstable territory at the moment, and a few (albeit dedicated) volunteers are not going to be sustainable long-term. We also need to do this to distance ourselves from both Audacity and especially Sneedacity - I don't want to push the flame war further, but TBH they are fiercely defensive and not welcoming at all when I tried to have a [polite conversation](https://github.com/Sneeds-Feed-and-Seed/sneedacity/issues/120) with them. I believe we can do better than that.
* The third thing is to establish a **dedicated identity for Tenacity**. Because, unfortunately, most users of Audacity use it for convenience (as it is *gratis* software) - they don't care at all about the free and open-source nature of it, and wouldn't care less if told about the privacy violations that lead to Tenacity's birth. If Tenacity is nothing but Audacity with a few minor differences, no one will use it. To deal with this issue, my recommendation would be to start making a **design system** for Tenacity. While the words "redesign" may sound quite terrifying, I plan on making incremental, small-scale UI changes that would slowly get users accustomed to a more polished and intuitive UI. This includes building a brand color palette, setting typography, and standardizing UI components. I think open-source design is a completely possible and worthy goal, and what better place to implement it than a community-centric program like Tenacity?

I believe that we should add a `ROADMAP.md` detailing our future direction, and add a `Development Plans` section in the README which would point to the `ROADMAP.md`. Any thoughts and criticisms on this idea would be greatly appreciated.

**Comment by fossdd** [comment]
You may also want to read https://github.com/tenacityteam/tenacity/issues/8

**Comment by Songtech-0912** [comment]
I understand there are major divisions in the team - that's understandable, Tenacity has had a difficult period recently. Additionally, the lack of a good platform for communicating live without javascript or sharing info on this area complicates matters. But we've got to work with what we have, don't we? I think the original points in the discussion hold true - a focus on building our identity as distinct from Audacity and as a community-based user-centric music application.

**Reply by unbeatable-101** [reply]
> Additionally, the lack of a good platform for communicating live without javascript or sharing info on this area complicates matters.

There are matrix and irc channels

**Reply by Songtech-0912** [reply]
Those work but aren't the most friendly for newer developers or newer users to get used to. Something like Mastodon and Telegram used together would be best in collaboration - but they don't solve one major issue, which is to have easily sharable knowledge bases and planning areas. Sites such as [notion](https://notion.so) and [HackMD](https://hackmd.io) are great for this type of thing but they are proprietary and require JavaScript.

**Reply by falkTX** [reply]
Matrix is literally opening a webpage, set user and password and be done with it...

It is not complicated and quite on par with other solutions.

**Reply by Songtech-0912** [reply]
Ok, thanks for letting me know, I'll do that right away. Additionally, I will also do the following:

* Register a Tenacity organization on [Penpot](https://penpot.app/), a free/libre alternative to Figma
* Begin working on a new website design on Penpot - this is important in establishing us as an independent and separate project
* Begin work on UI exploration and doing research on how the UI may be improved - it's good already but it could use some polish

**Reply by nbsp** [reply]
Using IRC can actually done from the Web, without an account. See [gamja IRC](https://web.libera.chat/gamja).

**Comment by Songtech-0912** [comment]
Alright, so I've made an initial draft design for the website. Here it is:

![image](https://user-images.githubusercontent.com/61605733/124998538-9a740b80-e01a-11eb-9b18-dac5c933f66f.png)

Feedback and criticisms on this would be appreciated.

**Reply by NotAProton** [reply]
Nice design. How heavy is it? Would you post your html and css?

**Reply by peepo5** [reply]
dark theme would look better imo :L

**Reply by peepo5** [reply]
also i would put features like- privacy-respecting, etc above download button

**Reply by Songtech-0912** [reply]
@not-a-proton It's meant to be super lightweight, no frameworks, just HTML + CSS, so < 200KB. However, I'm not super happy with the design. I think some further exploration is necessary.

**Reply by Gapstare** [reply]
If we are going with a bright theme, could we at least tone it down a bit? I have a problem looking at bright screens for an extended time.

**Comment by Songtech-0912** [comment]
Additionally, here's my design for a conceptual improved UI:

Of course, this is anything but complete and finished. Many parts of the UI should be improved, and there are big issues with the mockup. However, it is a good foundation to work on further.

![Tenacity UI Design](https://user-images.githubusercontent.com/61605733/125028096-666a0c00-e055-11eb-99c2-49971e20318f.png)

**Reply by peepo5** [reply]
Looks really good! Make sure to share this somewhere where wxwidgets and qt is discussed. if/when wxwidgets are replaced, a better interface such as this could be implimented.

**Reply by cookiengineer** [reply]
Did you use a specific iconset for creating the (smaller) buttons on the right (below the volume meters)? If a specific icon set can be used, it would ease up the spritemap generation part.

Audacity's codebase chose to use a crappy icon format (xpm) that's basically C-code representing a char array; and I'd love to make this use something like png (or if possible) svg formatted icons that would ease up the process of potential theming later.

**Reply by Songtech-0912** [reply]
The icons I used were a mix of 3 icon sets:
- 50% were from Material icons
- 30% were from Font Awesome (but this was because I was a bit too rushed to carefully look through the material icons and find the ones I wanted)
- 20% were custom-designed ones, you can see (a limited set) of them [here](https://github.com/Songtech-0912/tenacity-design)

@cookiengineer I'd love to help with the effort to clean up the icon system, especially because I'm actually also working on an icon set at the moment (from another project).

**Comment by Songtech-0912** [comment]
Additionally, I also have an idea for future governance of the project: we can assign a periodically rotated set of experienced contributors to serve as governance members of the project, keeping the project alive, moving, and in good hands.

**Comment by Songtech-0912** [comment]
New website design (updated) - do note that the text here is JUST PLACEHOLDER TEXT, I am a horrible writer and I just added whatever came to mind in terms of labels!

![image](https://user-images.githubusercontent.com/61605733/126054202-13b10b6c-96a1-4126-be10-01149e660320.png)

See https://github.com/tenacityteam/tenacity/issues/163#issuecomment-881988712 for more details

**Reply by mc776** [reply]
The light-themed one you posted just over a week ago was okay, but this just made my brain NOPE right out. It looks too slick, too commercialized, like someone's trying to sell me some BS and distract me from something that's going to be seriously problematic in the long (or short) run.

But of course when I saw one of your points in the OP one of my first thoughts was also "amateurs built the Ark, professionals built the Titanic" and how a lot of the best, most reliable FOSS projects are literal amateurs in the sense of people who won't or can't do this as a day job and are just putting their own scarce free time and work into something they believe in - the exact opposite kind of vibe this kind of web design gives.

**Reply by Songtech-0912** [reply]
@mc776 I must respectfully disagree with your conclusions. 

Granted, I understand where you are coming from:

* The "corporate" vibe is a problem, definitely, but that is only true **if the project isn't backed up by free/libre software and open source code!** And that's definitely not true for us.
* **We can be both professional _and_ believe in the work we do**. These 2 are not contrasting objectives. Think about the Python project, for instance - it is a very professional programming languages (with many powerful features) but its community is incredibly enthusiastic as well.
* I actually think this website is more inviting, but that's my perspective as a designer. A pure-text website is going to confuse users who want to actually see the features of software. A badly-designed website, one that is amateurish (if I understand correctly), is even worse - not just confusing users, but giving them a bad impression of the software itself.

On your point that the best FOSS projects are typically made by amateurs, there are many examples of widely adopted and reliable FOSS projects with amazing websites:

* <https://tailwindcss.com/>
* <https://nodejs.dev/>
* <https://www.blender.org/>

Besides, if "too slick, too commercialized" is your issue with the design, then that's basically discarding _all_ of today's design principles? My previous design had severe design flaws. It had poor visual hierarchy, poor typography, and a lack of sufficient contrast. This one still has problems, sure, but much fewer. If you just want to make a badly-designed website, I must respectfully disagree. If I have misunderstood what you meant, then feel free to show me a design that is both better than mine and satisfies the requirement of being not too "corporate".

**Reply by mc776** [reply]
To be honest? I *do* hate the vast majority of modern Web design. Why do we need to make sure the user wastes so much time scrolling through multiple screens for information that should have easily fit into a single screen at a perfectly legible font size, without these enormous headers and banner graphics and padded whitespace?

I have a visceral dislike for Tailwind's site and Node's just looks utterly vile. (Disclosure of bias: I am an avid user of NoScript. That lovely little three-word slogan already makes me want to burn something. To its utmost credit, however, I can at least *read* it without enabling Javascript or disabling CSS.)

For comparison:
https://www.dillo.org/
https://www.debian.org/

I like Dillo's site far more. It shows all the information you need in a no-nonsense, easily parsed manner. The only reason why I wouldn't say that all websites should be like this is that it's not mobile-degradable - the side panel stays exactly in place on my phone and everything is almost unreadably tiny.

Debian's is... more "modern", and in terms of layout is admittedly not too dissimilar to what you've got here. But I still like it a lot more than this, more than any difference in layout alone would explain - so I think the real difference here is **content**.

This "take back control" is already framing things in terms of conflict and fighting, like you're trying to rile up the reader against an enemy in order to sell someone on something. Granted, given the reasons we're all flocking to this fork there's a good reason for that, but to let that define the project's brand is to condemn it to giving permanent voice to the most toxic sorts of users later on.

And even the "take back" language itself combined with the aggressive red-white-black theme is very alienating for a lot of us who had to sit through some truly evil "take this country back" political posturing from the past 20 years.

Even without getting into the combativeness of the implications, the big bold imperative order at the reader is just... *rude*. Contrast Dillo, Debian and Blender, all of which first and foremost describe what they are, know the reader isn't an idiot and doesn't need to be *told to use it* if the description matches what they need. (There are imperative-sounding verbs on the Blender screen, but both of those - "Download" and "Get involved" - are labels on links and buttons, with the implicit "if you want to ___, click here".)

The description in the smaller text should have been the bigger text, without that rude angry warlike slogan. But even that doesn't really do justice to what the project is: a flexible, easy-to-use but powerful sound editor that lets you mix and create and convert all sorts of sound files. There's no reason to make the privacy the lede except in the context of Muse Group's telemetry - and if you're gonna start with that we might as well actually brand the project "We Are Not A Mused" or something and explicitly refer to all that drama right on the front page.

But that's not the value of the project. I just spent a good chunk of an evening earlier this week trying out different options instead of Audacity or any fork thereof to edit some sound clips for a game, and *nothing* I've found compares - compared to Audacity/Tenacity either it's like a decade obsolete, grossly inflexible or way too complicated or specialized for something I wasn't trying to do. So why not sell it on the positive stuff first, and leave the stuff about the Audacity/Muse drama to the FAQ?

**Reply by Songtech-0912** [reply]
All right, I get it. We have fundamental disagreements in opinions, perhaps exacerbated by the fact that I'm a UI/web designer. The wording ("copy text") can be changed anytime, I'm not suggesting it as the final text. It is probably **going to change** because my idea of the wording is definitely subject to change (I don't like the wording either but for other reasons).

There is one thing I do agree with you though - information density is way too low on most web pages, and aesthetics *has* been prioritized over content. If you want, we can discuss this issue in private - I am open to new viewpoints.

**Reply by bitsper2nd** [reply]
Sorry @mc776 but you have way too much bias over current design trends that it makes you miss the forest for the tree.

>It looks too slick, too commercialized, like someone's trying to sell me some BS and distract me from something that's going to be seriously problematic in the long (or short) run.

Nothing wrong with having a foss project that looks good. This like saying that poor people shouldn't have nice houses. If the user in this case likes the nice design, no harm done. They will feel better and more likely use Tenacity because of it. 

>To be honest? I do hate the vast majority of modern Web design. Why do we need to make sure the user wastes so much time scrolling through multiple screens for information that should have easily fit into a single screen at a perfectly legible font size, without these enormous headers and banner graphics and padded whitespace?

There is no final design yet. There is not even a Design System for Tenacity. There are wrifreframes, mockups, or any other kind of draft. And that is ok, because the best part of modern design when done good, is that you can make prototypes that lets users test some ideas. The designer and researcher get feedback from the users, and they can change the design.

>I have a visceral dislike for Tailwind's site and Node's just looks utterly vile. (Disclosure of bias: I am an avid user of NoScript. That lovely little three-word slogan already makes me want to burn something. To its utmost credit, however, I can at least read it without enabling Javascript or disabling CSS.)

that is a non-issue. if you disable javascript, expect for some functionality to not work. most users don't care. they leave javascript enabled.

**Reply by nm17** [reply]
Idk, seems kinda cool

**Reply by Garretiko** [reply]
I am completely in the same boat as @mc776. The slogan "Taking back control" conveys the wrong message. An audio editor isn't about "taking back control", it is about creating and manipulating sound files. The colours feel much to aggressive. And this so called "modern" web design looks mostly bland and empty and forces me to use my mouse much to much. Please consider a little more information density and less whitespace.

**Reply by Songtech-0912** [reply]
@Garretiko @mc776 Thanks for the feedback. I want to stress the idea that this is a *draft* of all things, and the copy text is just a placeholder (and I hadn't really considered what words I used). More info density will come, lots of people have pointed this out and it's my fault that I didn't consider how much I over-used spacing. I've already made these changes:
- Changed the colors to warmer colors
- Changed most of the text to actual placeholders (i.e. lorem ipsum idor...) as I don't want people to be confused by this

I will research ways to reduce the reliance on whitespace and increase info density. Please see https://github.com/tenacityteam/tenacity/issues/163#issuecomment-882042613 for more detail on the many in-progress changes.

**Reply by emabrey** [reply]
I think this design looks great, but I hope everyone understands that the criticism is welcome too. I personally just want the most people to have good free and open-source software in their hands as we possibly can. I think having an inviting and polished website helps achieve that, but I also agree that we don't want to be something we are not with regard to imitating corporately sterile designs. We should make sure our website has some personality. 

Also, please none of that "abstract people" corporate design language.  Please though, nothing like this:

![image](https://user-images.githubusercontent.com/3370875/126287573-12e49858-f1ca-45b0-866a-aea512d70539.png)

**Reply by Songtech-0912** [reply]
Haha, I have the same dislike for that sort of design :P You can guarantee that I won't use that.

And criticism is of course welcome! Designing is all about feedback, lots of feedback makes for an excellent design.

**Reply by bitsper2nd** [reply]
I would like to remind people that the best way for @Songtech-0912 to receive feedback directly is to open an account in Figma and comment on the mockup design. Here is the [latest iteration of the mockup](https://www.figma.com/proto/Y9bZR02MQHkiutuFkgyp3u/Tenacity-Website-Redesign?node-id=1%3A7&scaling=min-zoom&page-id=0%3A1). 

In case you don't want to open an account on figma(it's free as in beer), just leave a comment below then. The link above is to see the design as it were a live page for the website. It's what we call a prototype. Currently there is only a mockup for the home page. But as the design progresses, you will see the mockups for other pages. 

![Website V1](https://user-images.githubusercontent.com/22452474/126396157-24aa0ed6-abef-44f7-92ca-52a1bc4686d9.png)

**Reply by RAMChYLD** [reply]
> that is a non-issue. if you disable javascript, expect for some functionality to not work. most users don't care. they leave javascript enabled.

How about a compromise? One thing web devs seem to be forgetting nowadays is accessibility. We can have the javascript flashiness on the page meant for most browsers, but how about also having a more basic version of the page that is viewable in links/lynx, and the user is redirected to that page if the browser replies with a user agent that is not a js-capable browser?

**Reply by bitsper2nd** [reply]
> One thing web devs seem to be forgetting nowadays is accessibility.

Nope, modern web devs do consider[ accessibility](https://www.w3.org/standards/webdesign/accessibility). Javascript is another tool for that. 

> but how about also having a more basic version of the page that is viewable in links/lynx, and the user is redirected to that page if the browser replies with a user agent that is not a js-capable browser?

Then you should switch to a more capable browser that complies with the standards. Sorry.

**Reply by RAMChYLD** [reply]
The reason I brought up lynx is because it's upfront the easiest way to tell if the site will read properly to visually impaired people using screen readers or braille displays. Javascript often causes these to malfunction.

**Reply by yonderbread** [reply]
> The reason I brought up lynx is because it's upfront the easiest way to tell if the site will read properly to visually impaired people using screen readers or braille displays. Javascript often causes these to malfunction.

While I do agree it would be awesome to have a more minimalist version of the website available, modern HTML has [built-in features for accessibility](https://www.w3.org/TR/wai-aria-1.1/) such as screen readers.

**Comment by mhsabbagh** [comment]
Eagerly waiting for a release from your side folks, so that we can publish a press article about it on FOSS Post: https://fosspost.org

Fully supporting this.

**Comment by FrostKnight** [comment]
Glad you guys are working on this, it is extremely sad to see what audacity has become. 

One could say, them calling themselves open source is now an "audacity"

:p

That being said and pun intended, I would love it, if you could remove the conan dependency and instead use something less bloated.

Heh, but if you don't want to, that's fine, I am just extremely glad you forked from Audacity.  

Hope it all works out for you!

I hope yonderbread sees this repository, he was trying to maintain a fork, but the audacity team got his taken down.

Anywho, peace, hoping for you guys to be blessed to find a new way forward. :)

**Reply by poyribn** [reply]
> I hope yonderbread sees this repository

I just mentioned @yonderbread in this reply so he is much more likely to see it

**Reply by crsib** [reply]
He moved to GitLab: https://gitlab.com/yonderbread/audacity/activity

**Reply by yonderbread** [reply]
My apologies for not getting back to you guys sooner.

It seems that my decision to change the license from GPLv2 to v3 may have put the Audacity team's neck-bearded devs in a fit. Panicked, I backed up the code, deleted the Github fork and [put it up on Gitlab](https://gitlab.com/yonderbread/audacity) where it would be safe from these corporate demon spawns. It's entirely my mistake if I had a misunderstanding of the license agreement terms and I apologize to anyone who got confused on why the repo just yeeted itself out of existence.

I'm not sure what to do with my project now that there are quite a few others that have gained attention and are actively maintained. I despise redundancy, so the last thing I want to do is start doing what everyone else is doing unless I were to think of an original plan for my fork.

Is anyone still interested in tossing some ideas around for a neat audio tool? Let me know.

**Reply by yonderbread** [reply]
Ping @awslabspl @crsib

**Comment by FrostKnight** [comment]
I saw a response, about how you should "switch to a capable browser that complies with the standards in response, to people who wanted to be able to use the website on lynx and non corporate like browsers" 

Javascript is not secure, modern standards are questionable right now, due to google and webextensions, etc...

My point being,  if bitsper2nd's position is the official position of this fork's community, then you are very close to alienating me.

Don't get me wrong, I don't like the direction the regular audacity is going in, but, that's such a rude position to take.  

That is like saying, "Deal with it or get lost!"

I cannot abide in such foolishness, I may have to look elsewhere...

IF you change your mind, however, or at least are reasonable about this, I may change my mind about supporting you.  That being said, I don't know if you will. If you won't, later... :/

**Reply by bitsper2nd** [reply]
> I saw a response, about how you should "switch to a capable browser that complies with the standards in response, to people who wanted to be able to use the website on lynx and non corporate like browsers"

Yes. The majority of people are using webkit and blink based browsers. There is no need to support the 2 or 3 users on lynx or any other obscured text based browser. 

> Javascript is not secure, modern standards are questionable right now, due to google and webextensions, etc...

This is just for a homepage website. It's not a security nightmare.



> My point being, if bitsper2nd's position is the official position of this fork's community, then you are very close to alienating me.

The most important thing here is Tenacity. As in the program, not the website. Getting mad over something trivial such as website using javascript or not supporting lynx is pointless.

**Reply by yonderbread** [reply]
I say we keep the website as minimal as possible and make it entirely in pure HTML.

**Comment by FrostKnight** [comment]
To be honest @bitsper2nd, if you don't want to do something, that's fine I guess, but your reasoning is flawed as hell, its just... as unfriendly as can be.
Yes, it is a homepage, and no I don't use lynx, my reason for replying is just to point out, you were being very selfish with your reply more than anger or anything else...

PS, webkit and blink browsers suck, they are insecure and craploaded.

That being said, would it really be that much work to fix?

**Reply by bitsper2nd** [reply]
> PS, webkit and blink browsers suck, they are insecure and craploaded.

Those are the browsers that most users are going to be when accessing the website. Sorry, they are the standard. It is what it is.



> Yes, it is a homepage, and no I don't use lynx, my reason for replying is just to point out, you were being very selfish with your reply more than anger or anything else...

Sorry, but I'm being realistic with you. Supporting obscure browsers and removing javascript is not a Tenacity project goal.



> That being said, would it really be that much work to fix?

Why do you care more about a homepage website than the actual program that it was made for?

**Reply by akleja** [reply]
I've stayed out of the web design discussion so far. But I'd like to add my perspective.
* Base the page on sensible HTML written so it renders at least okay in text-based browsers and without stylesheets. This ensures accessibility. It is important for many screenreader based browsers to work, for one.
* Then add a layer of CSS to make it look fancy
* If javascript must be used, turning it off should not diminish accessibility, just remove som extra flair.

Accessibility has been a big focus in the design of the program for a very long time. Mostly making sure screen readers can make sense of everything, and having modes for the visually impaired. To me it would seem strange to not adhere to the same principles when designing the web page.

**Reply by bitsper2nd** [reply]
I agree with everything you just wrote. The discussion wasn't really much about design, but a disagreement with some users that are misinformed due to ignorance or simply being contrarian. They think that everything related to javascript and modern design trend is bad.

**Reply by akleja** [reply]
Then I guess I have nothing much to add.
Of course I have preferences and I guess I belong to the no-javascript crowd, but I can also see the benefit of mass appeal.

**Comment by FrostKnight** [comment]
Pointless comments have been deleted by Frostknight on this post.

Edit: This and most of my comments earlier this year were meaningless like grasping for the wind...

I wrote a reply elsewhere somewhere here... 

Feel free to delete all my comments from more than a week ago here if you want.  

As I may have mentioned below in a comment,  the web is getting increasingly bloated and non mainstream web browsers are gonna get shafted I fear, due to user agent blockades, javascript and other nonsense... also uxp sucks now because source code is being worked on behind closed doors somewhat, last I checked... Thus It is completely and totally prideful the way I thought before. Bitsper was absolutely right, although for a very different reason then at least I thought.
I did not foresee this insanity, kind of shows me how little I actually know about somethings... I posted twice btw on same topic just so he can see what I wrote... aka bitsper2nd.

**Reply by poyribn** [reply]
In the website redesign thread, [this](https://web.archive.org/web/20210803075301/https://sosasees.github.io/tenacity-webpage/) is this webpage I came up with.

Is it accessible? I believe so, because when I tested it on both my computer, my phone, and in lynx, all 3 web browsers displayed the page as intended.

It is based directly on the [Original tenacity website](https://web.archive.org/web/20210801221059/https://tenacityaudio.org/), because I thought it was already almost prefect.

**Comment by igavronski** [comment]
Any ideas on having an Android app? It would be great to have the ability to open projects on the road and do audio editing when traveling without carrying a laptop around!

**Reply by poyribn** [reply]
Tenacity is a computer program with lots of controls.
Despite being simple-to-use, tenacity is pretty complicated in construction.

I can't imagine this fitting on a phone. If somebody tried to do it anyway, the end result would likely be bad. Or if it's good it'll be completely different and won't have most of the functions of the computer version.

But the program could work very well on a Tablet, as their screens are large enough for the computer version's controls. Just make sure the touch targets are large enough to tap comfortably
(at least 48×48dp according to Material Design Guidelines;
1dp is basically 1 pixel except that dp are scaled with system scaling settings).

**Reply by igavronski** [reply]
Hi @Sosasees
I believe most road users wouldn't mind attaching either a mouse or keyboard to their phones to edit their audios. In fact, you don't need to conform to touch restrictions to use an app, am I right? In fact, just by having the ability to open your Tenacity projects from a cloud storage account and work on them, or extract audio from a video shot on your phone and applying a couple of effects (noise reduction, for example) would be great!
I believe a large community of phone/tablet/(?)Chromebook(?) users would benefit from the ability to do light edits while on the road.
All the best.

**Reply by Be-ing** [reply]
This is way beyond our capacity. wxWidgets doesn't support Android at all.

**Reply by bitsper2nd** [reply]
Sorry @igavronski, what you want is way out of scope for this kind of project. Phones and tablets are mainly consumption devices. They aren't meant to used to be productive like a Desktop Computer or Laptop. Also the making Tenacity for Android will require remaking the program almost from scratch. Let's not forget what @Sosasees wrote before. It would be really hard to make Tenacity for phones. You will have to sacrifice a lot of features to get the UX right on mobile. 

TL;DR: Tenacity for Mobile is an unrealistic high expectation.

**Reply by yonderbread** [reply]
We could try making Tenacity use a different UI library/framework besides wxWidgets and tweaking it to be more responsive, from there we could port it.

**Reply by poyribn** [reply]
If we changed þe widget framework, we'd have much more difficulty keeping up with upstream changes from Audacity.
So we can't afford changing þe widget framework yet.

We first need to release tenacity for desktop before we can even consider making a version for Android tablets.

**Reply by yonderbread** [reply]
Fair point and I agree we should get a working product done before anything else. If we were to migrate to a different framework, there may be a possibility of implementing some sort of translation layer for wxwidgets to convert calls to something it can interpret and render.

**Reply by bitsper2nd** [reply]
The first step would be to actually do a Design System for Tenacity.

**Reply by poyribn** [reply]
Let's start with þe colors, because any good design system has good colors.
¿How about þese ones based on [my tenacity webpage design](https://web.archive.org/web/20210809195638/https://sosasees.github.io/tenacity-webpage/)?:
![suggested tenacity colors by Sosasees](https://user-images.githubusercontent.com/66305550/129678401-5f3e1276-9945-40d2-983e-949cb8f729d6.png)

**Reply by bitsper2nd** [reply]
You mean the Color scheme? Check the [contrast accessibility](https://color.a11y.com/) beforehand. I would say do a prototype in figma with both dark theme and light theme. As for the website in your link, it looks good on mobile. On desktop the content below the main header would need to be changed.

**Reply by poyribn** [reply]
I have already checked þe color contrast when making my webpage design.
When I re-checked þe contrast, it's still AA accessible.

What exactly needs to be changed for desktop? I see nothing wrong.
You need to be more specific than "the content below the main header" before I undertsand.

**Reply by bitsper2nd** [reply]
Well, let me start with the obvious. Buttons are inconsistent. If they are supposed to serve as navigation, then a navbar would be a better choice. Two columns for the sections below the header is a must on desktop. Having only one column for is a lot of wasted space. See [mpv](https://mpv.io/) as a reference on how you can tweak the landing page for desktop. This is my feedback for now. You are free to follow through or ignore.

**Reply by poyribn** [reply]
For now þe website is good enough. Just like @bitsper2nd suggested,

> The first step would be to actually do a Design System for Tenacity.

, and then we can redesign the software and website for the better.

We will need a new discussion thread for the design system.

Update: I now made [the design system thread](https://github.com/tenacityteam/tenacity/discussions/487).

**Reply by yonderbread** [reply]
> Let's start with þe colors, because any good design system has good colors.
> ¿How about þese ones based on [my tenacity webpage design](https://web.archive.org/web/20210809195638/https://sosasees.github.io/tenacity-webpage/)?:
> ![suggested tenacity colors by Sosasees](https://user-images.githubusercontent.com/66305550/129678401-5f3e1276-9945-40d2-983e-949cb8f729d6.png)

Wouldn't it be better if the colors were variable and easily changeable with some sort of stylesheet or settings menu? Now if these are the default color schemes I think they look good but I feel like we should choose one that is reminicent of the old audacity colors, but with a vibrant fresh coat of paint.

**Reply by poyribn** [reply]
Yes: With how controversial the subject of the color palette can be if not done carefully, we need it to be user-changeable.

**Reply by yonderbread** [reply]
Something I'd really want to see done with this project is implenting features allowing for much more customization and UI layout arrangement within the app.

**Comment by Songtech-0912** [comment]
@ALL I would like to clarify a few things here:

- First, I do wish that everyone can stay peaceful and tolerant of a variety of opinions here. And everyone should be able to accept other people challenging and critiquing their viewpoints. Thanks to @mc776 I realized that my copy-writing was flawed (I've changed it to all placeholders now).
- I am wishing that @FrostKnight and @bitsper2nd can accept each others' viewpoints on the browser support issue. I am a user who uses BOTH the latest version of Firefox Developer Edition for webdev and Lynx/Gemini/Eww in Emacs to browse so I understand, to a varying level, the concerns of both of you. My position is to do both, we try to support everything as much as possible, and also make accessibility a top-level priority.
- Thank you to all for sharing commentary, I really do appreciate all of it. Please continue doing so.

**Reply by FrostKnight** [reply]
Yeah, I now feel certain, that the web is too bloated to use anything but mainstream web browsers or forks that support very new technology,
Let's just say, things have gotten very... toxic in the uxp web browser zone,  I am now realizing I was on a sinking ship...  more so, because the uxp devs have closed most of their source code thus making themselves appear to not care about open source, but that is neither here nor there,  I admit now, I was too prideful, forgive me,  also, I don't think github allows most web browsers that aren't mainstream anyways... so I don't see it as much of a problem as I did before. Perhaps bitsper2nd was right in at least one way, its just not sustainable for a different reason though, there is a lot of user agent blocking non mainstream web browsers happening... 

Long story short, I was too hotheaded and it was all meaningless...  -_- 
So yeah... sorry bitsper2nd and community fork of Tenacity, I hope Audacity has a force to reckon with, with at least one of these audacity forks. 

Anywho, peace!
